dstep/translator/CommentIndex.d

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: May 19, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.CommentIndex;
+
+import std.range;
+
+import clang.c.Index;
+import clang.Token;
+import clang.SourceLocation;
+import clang.SourceRange;
+import clang.TranslationUnit;
+
+class CommentIndex
+{
+    public struct Comment
+    {
+        string content;
+        SourceRange extent;
+        uint line;
+        uint column;
+        uint offset;
+
+        size_t indentAmount() const
+        {
+            import std.string;
+            import std.format;
+            import std.algorithm.comparison;
+            import std.algorithm.searching;
+
+            size_t amount = this.column - 1;
+            auto lines = lineSplitter(content);
+
+            if (!lines.empty)
+            {
+                lines.popFront();
+
+                foreach (line; lines)
+                {
+                    amount = min(
+                        amount,
+                        cast (size_t) countUntil!(a => a != ' ')(line));
+                }
+            }
+
+            return amount;
+        }
+
+        int opCmp(ref const Comment s) const
+        {
+            return offset < s.offset ? -1 : (offset == s.offset ? 0 : 1);
+        }
+
+        int opCmp(uint s) const
+        {
+            return offset < s ? -1 : (offset == s ? 0 : 1);
+        }
+    }
+
+    private TranslationUnit translUnit;
+    private Comment[] comments;
+    private SourceRange lastTokenRange;
+
+    this(TranslationUnit translUnit)
+    {
+        this.translUnit = translUnit;
+
+        lastTokenRange = translUnit.cursor.extent;
+
+        foreach (token; translUnit.cursor.tokens)
+        {
+            lastTokenRange = token.extent;
+
+            if (token.kind == CXTokenKind.CXToken_Comment)
+            {
+                auto location = token.location;
+
+                if (location.isFromMainFile)
+                {
+                    Comment comment = {
+                        token.spelling,
+                        token.extent,
+                        location.line,
+                        location.column,
+                        location.offset };
```

Comment:
Please avoid this initialization syntax. Either use `Comment(...)` or `{ key: value}`.